src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -436,6 +438,16 @@ module.exports = function<T, P, I, TI, C, CX>(
     } else {
       // Proceed under the assumption that this is a functional component
       workInProgress.tag = FunctionalComponent;
+      if (__DEV__) {
+        warning(
+          !!workInProgress.ref,
+          'Stateless function components cannot be given refs ' +
+          '(See ref "' + workInProgress.ref._stringRef + '" in ' +
+          getComponentName(workInProgress) + ' created by ' +
+          getComponentName(workInProgress.return) + '). ' +
```

Comment:
Perfect!
To be clear, the tasks are:
- [ ] update the current test case to have the stack addendum rather than just the owner name
- [ ] update both stack and fiber impls for that behavior.